pom.xml

```diff
@@ -261,6 +317,19 @@
             <groupId>org.sonatype.plugins</groupId>
             <artifactId>nexus-staging-maven-plugin</artifactId>
           </plugin>
+          <plugin>
+            <groupId>com.mycila</groupId>
+            <artifactId>license-maven-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>license</id>
+                <phase>compile</phase>
```

Comment:
why compile ? Maven has a convention that the build system should **never** modify any project file and this is the reason why generated files are put inside target/generated-sources.
The same way spotbug, checkstyle, etc are verification plugins working over the source files, license plugin is not exception and should be set on the verification phase, which is also how the plugin has been designed and how it is used pretty much everywhere.

putting it at compile phase would just make some sub projects force a  redefininition and move it to the verification phase and exclude it from compilation phase.